### PR TITLE
TRIP refactoring and add auth and access guards, and profile and home page providers

### PR DESCRIPTION
**Pull request by KbMstafa**

### CODE REVIEW

File: src/app/modules/auth/components/sign-in/sign-in.component.html [hunk 1]
```diff
@@ -2,6 +2,7 @@
   <img
     class="signin__logo"
     src="assets/images/logo-white.png"
+    alt="trip n tap logo"
```
Comment (Khousheish) [inline src/app/modules/auth/components/sign-in/sign-in.component.html]: even if its an alt add it to the translation json

File: src/app/modules/auth/components/sign-up/sign-up.component.html [hunk 2]
```diff
@@ -2,6 +2,7 @@
   <img
     class="signup__logo"
     src="assets/images/logo-white.png"
+    alt="trip n tap logo"
```
Comment (Khousheish) [inline src/app/modules/auth/components/sign-up/sign-up.component.html]: here also

File: src/app/modules/auth/store/auth.actions.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
-// tslint:disable: max-line-length
 import { createAction, props } from '@ngrx/store';
 
+// tslint:disable: max-line-length
```
Comment (Khousheish) [inline src/app/modules/auth/store/auth.actions.ts]: full file disables should be in the start of the file revert moving this file down

File: src/app/shared/types/module.types.ts [hunk 4]
```diff
@@ -1,2 +1,4 @@
 // tslint:disable: no-any
 export type Module = any;
```
Comment (Khousheish) [inline src/app/shared/types/module.types.ts:2]: this should be removed if were using modules type instead of it

Comment (KbMstafa) [reply src/app/shared/types/module.types.ts:2]: No we're using this also

**Review by Khousheish — APPROVED** [review]

**Pull request merged**